src/NServiceBus.Core/Performance/Metrics/MessagingMetricsFeature.cs

```diff
@@ -0,0 +1,39 @@
+namespace NServiceBus
+{
+    using System;
+    using Features;
+
+    /// <summary>
+    /// MessagingMetricsFeature captures messaging metrics
+    /// </summary>
+    class MessagingMetricsFeature : Feature
+    {
+        /// <inheritdoc />
+        protected internal override void Setup(FeatureConfigurationContext context)
+        {
+            var isSendOnly = context.Settings.GetOrDefault<bool>("Endpoint.SendOnly");
+            if (isSendOnly)
+            {
+                throw new Exception("Metrics are not supported on send only endpoints.");
+            }
+
+            RegisterBehavior(context);
+        }
+
+        static void RegisterBehavior(FeatureConfigurationContext context)
+        {
+            var settings = context.Settings.Get<ReceiveComponent.Settings>();
+            var endpointName = settings.EndpointName;
+            var discriminator = settings.EndpointInstanceDiscriminator;
+            var queueNameBase = settings.CustomQueueNameBase ?? endpointName;
```

Comment:
```suggestion
            var endpointName = context.Settings.EndpointName();
            var discriminator = context.Receiving.InstanceSpecificQueueAddress?.Discriminator;
            var queueNameBase = context.Receiving.QueueNameBase;
```

should use the values provided by the context.

I'm not sure we even need the endpoint name? Shouldn't be queuename + discriminator be valuable enough?

Reply:
Updated to use `context.Receiving`.